src/core/notifications/notificationsmanager.cpp

```diff
@@ -0,0 +1,225 @@
+#include "notificationsmanager.h"
+#include "core/bittorrent/session.h"
+#include "core/bittorrent/torrenthandle.h"
+#include "notifier.h"
+
+#include <QDir>
+#include <QUrl>
+
+#ifndef DISABLE_GUI
+#include "gui/mainwindow.h"
```

Comment:
This implementation is not good enough! (I didn't watch the rest), the Core MUST NOT in any way depend on a GUI (just don't tell me some of the remnants of such a code, I haven't cleaned it all).